packages/experimental/Menu/src/MenuItemCheckbox/MenuItemCheckbox.tsx

```diff
@@ -33,17 +34,27 @@ export const menuItemFinalRender = (
   menuItem: MenuItemCheckboxState,
   Slots: Slots<MenuItemCheckboxSlotProps>,
 ): React.FunctionComponent<MenuItemCheckboxProps> => {
-  return (final: MenuItemCheckboxProps) => {
-    const mergedProps = mergeProps(menuItem.props, final);
+  return (final: MenuItemCheckboxProps, children: React.ReactNode) => {
+    const { accessibilityLabel, ...mergedProps } = mergeProps(menuItem.props, final);
     const checkmarkXml = `
     <svg>
       <path fill='currentColor' d='M9.85355 3.14645C10.0488 3.34171 10.0488 3.65829 9.85355 3.85355L5.35355 8.35355C5.15829 8.54882 4.84171 8.54882 4.64645 8.35355L2.64645 6.35355C2.45118 6.15829 2.45118 5.84171 2.64645 5.64645C2.84171 5.45118 3.15829 5.45118 3.35355 5.64645L5 7.29289L9.14645 3.14645C9.34171 2.95118 9.65829 2.95118 9.85355 3.14645Z' />
     </svg>`;
 
+    let childText = '';
+    if (accessibilityLabel === undefined) {
+      React.Children.forEach(children, (child) => {
+        if (typeof child === 'string') {
+          childText = child; // We only automatically support the one child string.
+        }
+      });
+    }
+    const label = accessibilityLabel ?? childText;
+
```

Comment:
As this part of code is repeated, maybe it should be moved to a method?

Reply:
Yeah, good call